apps/content/schemaTypes/documents/hackathon-submission.ts

```diff
@@ -0,0 +1,90 @@
+import {defineField, defineType} from 'sanity'
+import {DocumentIcon} from '@sanity/icons'
+
+export const hackathonSubmission = defineType({
+  name: 'hackathonSubmission',
+  type: 'document',
+  title: 'Hackathon Submission',
+  icon: DocumentIcon,
+  fields: [
+    defineField({
+      name: 'hackathon',
+      type: 'reference',
+      to: [{type: 'hackathon'}],
+      title: 'Hackathon',
+      description: 'The hackathon this submission is for',
+      validation: (Rule) => Rule.required().error('Hackathon is required'),
+    }),
+    defineField({
+      name: 'person',
+      type: 'reference',
+      to: [{type: 'person'}],
+      title: 'Submitter',
+      description: 'The person who submitted this entry',
+    }),
+    defineField({
+      name: 'email',
+      type: 'string',
+      title: 'Email',
+      validation: (Rule) => Rule.required().email().error('Valid email is required'),
+    }),
+    defineField({
+      name: 'fullName',
+      type: 'string',
+      title: 'Full Name',
+      validation: (Rule) => Rule.required().error('Full name is required'),
+    }),
+    defineField({
+      name: 'githubRepo',
+      type: 'url',
+      title: 'GitHub Repository',
+      description: 'Link to the source code repository',
+      validation: (Rule) => Rule.required().uri({scheme: ['https']}).error('GitHub repo URL is required'),
+    }),
+    defineField({
+      name: 'deployedUrl',
+      type: 'url',
+      title: 'Deployed URL',
+      description: 'Link to the deployed web app',
+      validation: (Rule) => Rule.required().uri({scheme: ['http', 'https']}).error('Deployed URL is required'),
+    }),
```

Comment:
I don't think we'll be able to make this required anymore. The apps are all using AI now so people can't deploy them for free. Sucks, but seems to be How Things Are™ for now.